cpp/unreal_plugins/SpEngine/Source/SpEngine/SpEngine.cpp

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "SpEngine/SpEngine.h"
+
+#include <memory>                  // std::make_unique, std::unique_ptr
+
+#include <Modules/ModuleManager.h> // IMPLEMENT_MODULE
+
+#include "SpCore/Assert.h"
+#include "SpCore/Config.h"
+#include "SpCore/Log.h"
+#include "SpCore/Rpclib.h" // rpc_server
+#include "SpCore/Unreal.h"
+#include "SpEngine/EngineService.h"
+#include "SpEngine/GameWorldService.h"
+#include "SpEngine/LegacyService.h"
+
+void SpEngine::StartupModule()
+{
+    SP_LOG_CURRENT_FUNCTION();
+    SP_ASSERT(FModuleManager::Get().IsModuleLoaded(Unreal::toFName("SpCore")));
+    SP_ASSERT(FModuleManager::Get().IsModuleLoaded(Unreal::toFName("UrdfRobot")));
+    SP_ASSERT(FModuleManager::Get().IsModuleLoaded(Unreal::toFName("Vehicle")));
+
+    if (Config::s_initialized_) {
+        rpc_server_ = std::make_unique<rpc::server>(Config::get<int>("SP_ENGINE.PORT"));
+    } else {
+        rpc_server_ = std::make_unique<rpc::server>(30000);
+    }
+    SP_ASSERT(rpc_server_);
+
+    // bind functions here to be consistent with the usage of these functions on the python side
+    rpc_server_->bind("sp_engine.ping", []() -> std::string {
+        return "SpEngine received a call to ping()...";
+    });
+
+    rpc_server_->bind("sp_engine.request_close", []() -> void {
+        bool immediate_shutdown = false;
+        FGenericPlatformMisc::RequestExit(immediate_shutdown);
+    });
+
+    // EngineService needs its own custom logic for binding its entry points, because they are
```

Comment:
Great comment. Can any of the services be objects instead of pointers? I remember we worked through the details together but I can't remember the conclusions.

Reply:
`EngineService` cannot be an object. However, I think `GameWorldService` and `LegacyService` can be. But, to maintain consistency across the file, I used pointers for both of them as well.

Reply:
Good choice.